specs/data-grid/data-grid.spec.md

```diff
@@ -0,0 +1,302 @@
+
+*properties:*
+- `rowsData`  
+An array of objects that contain the data to be displayed.  Each object corresponds to one row.
+
+- `columnsData`  
+An array of `DataGridColumn` objects that define what columns will be displayed in the grid.  The order of the columns determines their order in the grid.
+
+- `rowItemTemplate`  
+Custom [template](https://fast.design/docs/fast-element/declaring-templates) to use when generating rows by iterating over data.  The default template uses `fast-data-grid-row`, this is where authors change that.
+
+*Slots:*
+- `default`  
+Custom generated rows can be placed here
+
+- `header`
+Items added here are added to the header region of the component.
+
+*Functions:*
+- `generateColumns(object): DataGridColumn`   
+Static function that creates a basic set of columns from an object representing a row.  
+
+*Events*
+- none
+
+*parts:*
+- `headerSlot`
+
+- `rowsSlot`
+
+
+**Data grid header**
+- `fast-data-grid-header`
+
+*Attributes:*
+- none
+
+*properties:*
+- `columnsData`  
+An array of `DataGridColumn` objects that define what columns will be displayed in the grid.  The order of the columns determines their order in the grid.
+
+- `headerCellItemTemplate`  
+Custom [template](https://fast.design/docs/fast-element/declaring-templates) to use when generating header cells by iterating over data.  The default template uses `fast-data-grid-header-cell`, this is where authors change that.
+
+*Slots:*
+- `default`  
+Default slot for items
+
+*Events*
+- none
+
+*parts:*
+- `headerCellsSlot`
+
+
+**Data grid header cell**
+- `fast-data-grid-header-cell`
+
+*Attributes:*
+- extends HTML Element attributes
+
+*properties:*
+- `columnData`  
+The `DataGridColumn` this column header represents.
+
+*Slots:*
+- `default`  
+Default slot for items
+
+*Events*
+- none
+
+*parts:*
+- `headerCellsSlot`
+
+
+**Data grid row**
+- `fast-data-grid-row`
+
+*Attributes:*
+- `gridTemplateColumns`  
+String that gets applied to the the css gridTemplateColumns attribute for the row
+
+*properties:*
+- `rowData`  
+The objects that contains the data to be displayed in this row.
+
+- `columnsData`  
+An array of `DataGridColumn` objects that define what columns will be displayed in the grid.  The order of the columns determines their order in the grid.
+
+- `cellItemTemplate`  
+Custom [template](https://fast.design/docs/fast-element/declaring-templates) to use when generating cells by iterating over data.  The default template uses `fast-data-grid-cell`, this is where authors change that.
+
+*Slots:*
+- `default`  
+Default slot for items
+
+*Events*
+- none
+
+*parts:*
+- `cellsSlot`
+
+**Data grid cell**
+*Component name:*
+- `fast-data-grid-cell`
+
+*Attributes:*
+- `gridColumnIndex`  
+The grid column this cell appears in.
+
+*properties:*
+- `rowData`  
```

Comment:
Is the cell supposed to have row data?

Reply:
same question about columnData. Wouldn't this represent an item in a row and column?

Reply:
Yes, cells should have rowdata.  There is not necessarily a 1:1 relationship between columns a specific property in the row data.  For example a grid display could have a column that looks at multiple values in the data row to determine what it displays, such a green checkmark if a student has completed all the assignments represented by multiple items in the rowdata.   This would be done with a custom cell template specified in the column data.

Reply:
As for columnData, that defines how a particular column should be rendered but does not reference the actual data beyond what property keys to look for.

Reply:
In the React version we used "column definition" instead of "column data" which may have made this difference clearer. I've been vacillating about whether to use that terminology again.